src/components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,133 @@
+"use client";
+
+import React, { useState } from "react";
+import FullCalendar from "@fullcalendar/react";
+import timeGridPlugin from "@fullcalendar/timegrid";
+import dayGridPlugin from "@fullcalendar/daygrid";
+import interactionPlugin, { DateClickArg } from "@fullcalendar/interaction";
+import { EventClickArg } from "@fullcalendar/core";
+import styled from "styled-components";
+import { Dialog } from "@mui/material";
+import { EventImpl } from "@fullcalendar/core/internal";
+
+interface CalendarProps {
+    initialEvents: CalendarEvent[];
+    view?: string;
+    editable?: boolean;
+    handleDateClick?: (info: DateClickArg) => void;
+}
+
+export interface CalendarEvent {
+    id: string;
+    title: string;
+    start: Date;
+    end: Date;
+    description?: string;
+    daysOfWeek?: number[];
+    startRecur?: string;
+    endRecur?: string;
+    backgroundColor?: string;
+    borderColor?: string;
+    textColor?: string;
+    allDay?: boolean;
+}
+
+const StyledDialog = styled(Dialog)`
```

Comment:
Yep probably, that can be unified in another ticket as ours doesn't look exquisite either.

Reply:
I think they are also being used in data-view component. But they are done with react-modal instead of dialog. I think we should make it a separate component (using either dialog or react-modal for consistency). @ChloeYiuSoftwire @paopaosoftwire 

I think dialog would be better since mui is already used in the icons component.